### DISCUSSION TITLE
How ai works

### DISCUSSION THREAD

**Discussion by Pardeep-Kumar0** [community/community#84097] [Programming Help]
Could you explain how ai works ? How ai give information whatever we want

**Comment by sudhirhub** [comment]
how to ai work to do any thing..............????

**Comment by ghost** [comment]
Hi @Pardeep-Kumar0 


Imagine you want to teach a robot to recognize fruit. You would show it lots of apples, oranges, and bananas, telling it, "This is an apple, that's an orange, and this is a banana". The robot takes pictures and notes details about each fruit, like color and shape. Then, when you show it a new fruit, it looks at those details and guesses what it is based on what it's learned before.

AI works in a similar way. It learns from examples or data, finds patterns in that information, and then uses those patterns to make decisions or predictions. Whether it's recognizing faces in photos or helping recommend movies you might like, AI uses what it's learned to do helpful things.

AI works by recognizing patterns and trends from a given data/task and then it implements the things it learned from the training data that the engineers had given during training stage.

In simpler terms, imagine that giving your friend a bunch of examples of something, like different types of dogs. You show them pictures and explain what makes each dog different. After seeing lots of examples, your friend learns to recognize different dogs by their features. AI does something similar but with tons of data and math.

> [!Note]
> There also exists some AIs that work on rule based approach. So this explanation is common for all types of AI but not at 100%

I hope I was helpful to you....


Thank You

**Comment by Ybouali** [comment]
u can say it is works like human brain but is not because we still trying to understand how does human brain works
More explanation: 
human brain learn from experience and what hi know ,
if we take a child and keep telling to this child the life is too easy don't do nothing mony will comes to u just keep playing the child will not be a someone ho will  is a strong in this life ,
but if the child know what is the real life so we will found a new brain to discover

**Reply by Brooksie18** [reply]
I agree 100%

**Comment by cosmicqbit** [comment]
Check out [this video](https://www.youtube.com/watch?v=PHNJ2_4oefE) on Youtube

**Reply by ghost** [reply]
Nas daily vid was awesome...

**Comment by ghost** [comment]
## Basic Diagram for functioning of Ai
![47-3 (1)](https://github.com/community/community/assets/121353576/7cafbbc8-0a8a-488f-a61b-7747760bee5f)

## AI Subsets
![how_ai_works-f_mobile](https://github.com/community/community/assets/121353576/c4d83078-f4db-4928-8644-489e4aa9bea1)

## Various stages in AI 
![5_KI_Wie_funktioniert_maschinelles_Lernen_190_05_OER_E](https://github.com/community/community/assets/121353576/11a3165f-ce05-4fa8-a1fd-0b440d2ab789)

## Detailed Explanation

![how_does_explainable_ai_work_ppt_powerpoint_presentation_file_designs_slide01](https://github.com/community/community/assets/121353576/047fe3da-56f1-4f72-b2d8-39ed354c60a2)


> [!Note]
> I had checked these visual representation of AI functioning and thought this might help.I Hope it helped you...

Thank You

**Comment by Pardeep-Kumar0** [comment]
Wow! Bro we appreciate you . 
I like the way you deeply understand us how ai functions.
It's really helpful for us .
Thanks for it

**Reply by ghost** [reply]
Your Welcome ❤️

**Reply by Pardeep-Kumar0** [reply]
Hey @Yuheshpandian bro I don't know how to use GitHub properly ! Could you please tell me Can we learn from GitHub i meant there is any best  programming courses or anything else who help me to learn coding if there is courses so how can take access of them

**Reply by ghost** [reply]
Yes of course I could....

**Reply by Pardeep-Kumar0** [reply]
How to access them?

**Reply by ghost** [reply]
# Sources To Learn Github:
- **[Tutorial](https://youtu.be/RGOj5yH7evk?si=Kg9CFWtTGh3PKbB3)** from free code camp. I would suggest this the most to learn about Github.
- Using GitHub effectively involves understanding version control, repositories, and collaboration. Start by creating a repository for your project, regularly commit changes with clear and descriptive messages, and create branches for new features or fixes. Pull and merge changes frequently to stay up-to-date with the main branch, and use issues and pull requests for effective communication and code review among collaborators. Document your code, follow best practices, and leverage GitHub's features like Actions and Projects to automate workflows and manage tasks efficiently.

# Best Platform for Courses :
- [Free Code Camp](https://www.freecodecamp.org/) : The best Free one as there is a huge community.
- [Coursera](https://www.coursera.org/courses?query=free) : It offers some best free Courses that are updated to current.
- Udemy courses are also in good quality but I never tried one myself.
- I had been using Sololearn to learn Python and Html but now changed my mind.
- For web dev I would suggest courses from [Dave Gray](https://www.youtube.com/@DaveGrayTeachesCode): He is a Well Experienced Web Dev who creates Youtube Tutorial.

> [!Tip]
> Whatever course you choose instead on falling into tutorial hell, try to practice the stuff you learnt on that moment itself.

**Reply by ghost** [reply]
That's all I would suggest for now. But try to practice learning from Web sources instead of tutorials. A certificate doesn't define you efficiency completely...

**Comment by samus-aran** [comment]
Hi @Pardeep-Kumar0 , Thanks for joining us for the GitHub Universe celebrations in the GitHub Community, we're glad you're here! 

We want to make sure you’re getting the best support you can now that GitHub Universe has ended so I've gone ahead and moved this to the most relevant category for you.

Thank you for helping us maintain a productive and tidy community for all our members.
